UI/Controllers/HRController.cs

```diff
@@ -49,17 +52,29 @@ public CandidateDTO Get(int id)
         
         // POST api/<HRController>
         [HttpPost("InterviewResults")]
-        public async Task Post([FromBody] HRInterviewResults hrInterviewresult) //it doesn't work correctly, need to change
+        public async Task Post([FromBody] HRInterviewResults hrInterviewResultsUI)
         {
-            HRInterviewDTO hRInterview = _mapper.Map<HRInterviewDTO>(hrInterviewresult);
+            HRInterviewDTO hRInterview = _mapper.Map<HRInterviewDTO>(hrInterviewResultsUI);
             var employee = await GetEmployee();
             hRInterview.EmployeeID = employee.Id;
-            _interviewService.AddHRInterview(hRInterview);
+            hRInterview.ID = _interviewService.AddHRInterview(hRInterview);
+
+            var skillKnowledgeDTOList = _mapper.Map<IEnumerable<SkillKnowledgeDTO>>(hRInterview);
+            _skillKnowledgeService.AddSkillKnowledge(skillKnowledgeDTOList);
         }
 
         [HttpPost("InterviewResultsWithDeclineStatus")]
-        public void Post([FromBody] HRInterviewDTOWithDecline hrInterviewDTODecline)
-        { }
+        public async Task Post([FromBody] HRInterviewDTOWithDecline hrInterviewDTODecline)
+        {
+            HRInterviewDTOWithDecline hRInterviewWithDecline = _mapper.Map<HRInterviewDTOWithDecline>(hrInterviewDTODecline);
+            var employee = await GetEmployee();
+            hRInterviewWithDecline.EmployeeID = employee.Id;
+            //hRInterviewWithDecline.StatusID
+            hRInterviewWithDecline.ID = _interviewService.AddHRInterview(hRInterviewWithDecline);
+
+            var skillKnowledgeDTOList = _mapper.Map<IEnumerable<SkillKnowledgeDTO>>(hRInterviewWithDecline);
+            _skillKnowledgeService.AddSkillKnowledge(skillKnowledgeDTOList);
+        }
```

Comment:
I don't see differences between InterviewResults and InterviewResultsWithDeclineStatus

Reply:
I am too, but I don't know how I can use endpoint method